generator/mapper.go

```diff
@@ -92,21 +58,36 @@ func getMappedCatalogControlsFromImport(importedCatalog *catalog.Catalog, profil
 		}
 		//Append controls to the new group if matches
 		for _, catalogControl := range group.Controls {
-			for _, z := range profileImport.Include.IdSelectors {
+			for controlIndex, z := range profileImport.Include.IdSelectors {
 				if strings.ToLower(catalogControl.Id) == strings.ToLower(z.ControlId) {
-					newGroup.Controls = append(newGroup.Controls, catalog.Control{
+					newControl := catalog.Control{
 						Id:          catalogControl.Id,
 						Class:       catalogControl.Class,
 						Title:       catalogControl.Title,
 						Subcontrols: []catalog.Subcontrol{},
-					})
+					}
+					//For subcontrols, find again in entire profile
+					for _, sc := range catalogControl.Subcontrols {
```

Comment:
`sc` should be replaced with more meaningful name.

Reply:
I believe thats how they code in golang, brevity over clarity

Reply:
yeah, you'll see a lot of "m" letter in orca as well. 
still changing it though